### PR TITLE
Add 'md5' function to PostgresModule

### LINKED ISSUE #189: Add 'md5' function to PostgresModule
Add support for function `md5` described in the PostgreSQL [docs](https://www.postgresql.org/docs/current/functions-string.html).
1. Add function to `PostgresFunctionDef` in `PostgresModule`
2. Add test in `FunctionDefSpec`

### PR DESCRIPTION
**Pull request by riccardocorbella**
Fixes #189

### CODE REVIEW

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 1]
```diff
@@ -108,7 +108,9 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
-    implicit def literal[A: TypeTag](a: A): Expr[Features.Literal, Any, A] = Expr.Literal(a)
+    implicit def literal[A](a: A)(implicit typeTag: TypeTag[A]): Expr[Features.Literal, Any, A] = typeTag match {
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/expr.scala:111]: Why this change?

Comment (riccardocorbella) [reply core/jvm/src/main/scala/zio/sql/expr.scala:111]: It's a typo! My bad...

**Review by jczuchnowski — APPROVED** [review]
